owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt

```diff
@@ -0,0 +1,142 @@
+import com.owncloud.android.domain.files.model.MIME_PREFIX_IMAGE
+import com.owncloud.android.testutil.OC_FILE
+import io.mockk.every
+import io.mockk.spyk
+import io.mockk.verify
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+
+class OCLocalFileDataSourceTest {
+    private lateinit var localDataSource: OCLocalFileDataSource
+    private lateinit var dao: FileDao
+
+    private val mapper = OCFileMapper()
+
+    private val dummy = OC_FILE.copy(id = 0)
+    private val dummyEntity = mapper.toEntity(OC_FILE)!!
+
+    @Before
+    fun init() {
+        dao = spyk()
+        localDataSource = OCLocalFileDataSource(dao, mapper)
+    }
+
+    @Test
+    fun `get file by id - ok`() {
+        every { dao.getFileById(any()) } returns dummyEntity
+
+        val result = localDataSource.getFileById(dummy.id!!)
+
+        assertEquals(dummy, result)
+
+        verify { dao.getFileById(dummy.id!!) }
+    }
+
+    @Test
+    fun `get file by id - ok - null`() {
+        every { dao.getFileById(any()) } returns null
+
+        val result = localDataSource.getFileById(dummy.id!!)
+
+        assertEquals(null, result)
+
+        verify { dao.getFileById(dummy.id!!) }
+    }
+
+    @Test(expected = Exception::class)
+    fun `get file by id - ko`() {
+        every { dao.getFileById(any()) } throws Exception()
+
+        localDataSource.getFileById(dummy.id!!)
+    }
+
+    @Test
+    fun `get file by remote path - ok`() {
+        every { dao.getFileByOwnerAndRemotePath(any(), any()) } returns dummyEntity
+
+        val result = localDataSource.getFileByRemotePath(dummy.remotePath, dummy.owner)
+
+        assertEquals(dummy, result)
+
+        verify { dao.getFileByOwnerAndRemotePath(dummy.owner, dummy.remotePath) }
+    }
+
+    @Test
+    fun `get file by remote path - ok - null`() {
+        every { dao.getFileByOwnerAndRemotePath(any(), any()) } returns null
+
+        val result = localDataSource.getFileByRemotePath(dummy.remotePath, dummy.owner)
+
+        assertEquals(null, result)
+
+        verify { dao.getFileByOwnerAndRemotePath(dummy.owner, dummy.remotePath) }
+    }
+
+    @Test(expected = Exception::class)
+    fun `get file by remote path - ko`() {
+        every { dao.getFileByOwnerAndRemotePath(any(), any()) } throws Exception()
+
+        localDataSource.getFileByRemotePath(dummy.remotePath, dummy.owner)
+    }
+
+    @Test
+    fun `get folder content - ok`() {
+        every { dao.getFolderContent(any()) } returns listOf(dummyEntity)
+
+        val result = localDataSource.getFolderContent(dummy.id!!)
+
+        assertEquals(listOf(dummy), result)
+
+        verify { dao.getFolderContent(dummy.id!!) }
+    }
+
+    @Test(expected = Exception::class)
+    fun `get folder content - ko`() {
+        every { dao.getFolderContent(any()) } throws Exception()
+
+        localDataSource.getFolderContent(dummy.id!!)
+    }
+
+    @Test
+    fun `get folder images - ok`() {
```

Comment:
Why do you sometimes write `- ok` and sometimes `- ko`
Just a question of interest: what is the scheme for writing the description/names of the tests?

Reply:
I use `ok` as success and `ko` as exception or failure.

I started writing names as `getFolderContentSuccess` or `getFolderContentException` but it is not as simple when you have several cases. So i decided to switch to a more readable way:

what are we testing - expected behaviour - additional info